sdk/monitor/opentelemetry-exporter-azuremonitor/tests/test_base_exporter.py

```diff
@@ -0,0 +1,417 @@
+        with self.assertRaises(TypeError):
+            BaseExporter(something_else=6)
+
+    def test_telemetry_processor_add(self):
+        base = self._base
+        base.add_telemetry_processor(lambda: True)
+        self.assertEqual(len(base._telemetry_processors), 1)
+
+    def test_telemetry_processor_clear(self):
+        base = self._base
+        base.add_telemetry_processor(lambda: True)
+        self.assertEqual(len(base._telemetry_processors), 1)
+        base.clear_telemetry_processors()
+        self.assertEqual(len(base._telemetry_processors), 0)
+
+    def test_telemetry_processor_apply(self):
+        base = self._base
+
+        def callback_function(envelope):
+            envelope.data.base_type += "_world"
+
+        base.add_telemetry_processor(callback_function)
+        envelope = TelemetryItem(
+            name="", time="", data=MonitorBase(base_type="type1"))
+        base._apply_telemetry_processors([envelope])
+        self.assertEqual(envelope.data.base_type, "type1_world")
+
+    def test_telemetry_processor_apply_multiple(self):
+        base = self._base
+        base._telemetry_processors = []
+
+        def callback_function(envelope):
+            envelope.data.base_type += "_world"
+
+        def callback_function2(envelope):
+            envelope.data.base_type += "_world2"
+
+        base.add_telemetry_processor(callback_function)
+        base.add_telemetry_processor(callback_function2)
+        envelope = TelemetryItem(
+            name="", time="", data=MonitorBase(base_type="type1"))
+        base._apply_telemetry_processors([envelope])
+        self.assertEqual(envelope.data.base_type, "type1_world_world2")
+
+    def test_telemetry_processor_apply_exception(self):
+        base = self._base
+
+        def callback_function(envelope):
+            raise ValueError()
+
+        def callback_function2(envelope):
+            envelope.data.base_type += "_world2"
+
+        base.add_telemetry_processor(callback_function)
+        base.add_telemetry_processor(callback_function2)
+        envelope = TelemetryItem(
+            name="", time="", data=MonitorBase(base_type="type1"))
+        base._apply_telemetry_processors([envelope])
+        self.assertEqual(envelope.data.base_type, "type1_world2")
+
+    def test_telemetry_processor_apply_not_accepted(self):
+        base = self._base
+
+        def callback_function(envelope):
+            return envelope.data.base_type == "type2"
+
+        base.add_telemetry_processor(callback_function)
+        envelope = TelemetryItem(
+            name="", time="", data=MonitorBase(base_type="type1"))
+        envelope2 = TelemetryItem(
+            name="", time="", data=MonitorBase(base_type="type2"))
+        envelopes = base._apply_telemetry_processors([envelope, envelope2])
+        self.assertEqual(len(envelopes), 1)
+        self.assertEqual(envelopes[0].data.base_type, "type2")
+
+    # def test_transmission_nothing(self):
```

Comment:
Are these changes being tracked anywhere?

Reply:
Tracking in our team backlog

Reply:
What is the guideline of submitting issues? Should we be using Github our Azure devops? @rakshith91 
Any guidance on submitting issues specific to azure monitor?

Reply:
Please log all the issues in github and use two labels - monitor and client

Reply:
@hectorhdzg 
Can you submit an issue to track this to github as well?

Reply:
https://github.com/Azure/azure-sdk-for-python/issues/15081